packages/teams-js/src/public/search.ts

```diff
@@ -0,0 +1,131 @@
+import { sendMessageToParent } from '../internal/communication';
+import { registerHandler, removeHandler } from '../internal/handlers';
+import { ensureInitialized } from '../internal/internalAPIs';
+import { errorNotSupportedOnPlatform, FrameContexts } from './constants';
+import { runtime } from './runtime';
+
+/**
+ * Allows you to interact with the host search experience
+ *
+ * @beta
+ */
+export namespace search {
+  const onChangeHandlerName = 'search.queryChange';
+  const onClosedHandlerName = 'search.queryClosed';
+  const onExecutedHandlerName = 'search.queryExecute';
+
+  /**
+   * This interface contains information pertaining to the search term in the host search experience
+   */
+  export interface SearchQuery {
+    /** The current search term in the host search experience */
+    searchTerm: string;
+
+    // TODO: may need some sort of timestamp or sequence value to
+    // ensure messages are processed in correct order / combine them
+    // having any sort of logic around combining messages or sorting them
+    // would make sense to go into the teamsjs-sdk layer
+    // timestamp: number;
+  }
+
+  export type SearchQueryHandler = (query: SearchQuery) => void;
+
+  /**
+   * Allows the caller to register for various events fired by the host search experience.
+   * Calling this function will cause the host search experience to set its default scope to
+   * the name of your application.
+   * 
+   * Your application should *not* re-render inside of these callbacks, there may be a large number
+   * of onChangeHandler calls if the user is changing the searchQuery rapidly.
+   *
+   * @param onChangeHandler - This handler will be called when the user begins searching and every
+   * time the user changes the contents of the query. The value of the query is the current term
+   * the user is searching for. Should be used to put your application into whatever state is used
+   * to handle searching. This handler will be called once with an empty {@link SearchQuery.searchTerm}
+   * when search is beginning. 
```

Comment:
This "will be called once with an empty searchTerm" part is interesting. Does that mean it'll only ever be called with an empty search term once? Or that it'll only be called with an empty search term at the beginning of user interacting with the searchbox? What if I start typing, and then backspace back to an empty term? Would we call this again with an empty search box? That seems like more than "once" to me?

Reply:
Yes sorry I could/should have been more clear. Every time the contents of the searchbox change this function will get called with the current contents. *In addition* it will also be called once when search begins (in which case search should be empty). If you type something and then delete it all this will be called again with the empty string.